core/src/main/java/org/elasticsearch/index/query/ConstantScoreQueryBuilder.java

```diff
@@ -66,20 +70,78 @@ public ConstantScoreQueryBuilder boost(float boost) {
         return this;
     }
 
+    /**
+     * @return the boost factor
+     */
+    public float boost() {
+        return this.boost;
+    }
+
     @Override
     protected void doXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject(NAME);
         builder.field("filter");
-        filterBuilder.toXContent(builder, params);
+        queryBuilder.toXContent(builder, params);
+        builder.field("boost", boost);
+        builder.endObject();
+    }
+
+    @Override
+    public Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException {
+        Query innerFilter = queryBuilder.toQuery(parseContext);
+        if (innerFilter == null ) {
+            // no inner query, we ignore this and return no docs
+            return Queries.newMatchNoDocsQuery();
```

Comment:
Seeing this is all over the place, including the DSL, I also think this should be changed at some point but it needs to change on all levels then. The way it seems we have to handle null-return values for any nested query at any point. Which brings me to the validate() method where I currently check for inner builder beeing null. I think I might need to revert that one as well as the above example doesn't throw an error in the current version of the DSL.